SMP minimal configurations fail to build

### Describe the bug

Test case `sample.minimal.riscv.runtime` fails to build on SMP RISC-V configuration.

After investigating the build failure, the root cause was traced to this commit https://github.com/zephyrproject-rtos/zephyr/commit/0111cc903031fd32bb8187c4ee44ddb125dcbb5b that introduced IPI work functionality that forces `CONFIG_EVENTS` to be enabled through `CONFIG_SCHED_IPI_SUPPORTED` which is automatically enabled on SMP-capable processor, but the events implementation in `kernel/events.c` unconditionally calls timeout APIs that require `CONFIG_SYS_CLOCK_EXISTS` to be enabled, but minimal runtime configurations explicitly disable system clock support **to reduce footprint**, but the IPI infrastructure now mandates event objects which assume timer infrastructure is always available.

So, I suggest to add conditional guard around timeout operation in kernel/events.c

### Regression

- [x] This is a regression.

### Steps to reproduce

west build -p -b qemu_riscv32/qemu_virt_riscv32/smp samples/basic/minimal -T sample.minimal.riscv.runtime

### Relevant log output

```shell
.../zephyrproject/zephyr/kernel/events.c: In function 'event_walk_op':
.../zephyrproject/zephyr/kernel/events.c:139:17: warning: implicit declaration of function 'z_abort_timeout'; did you mean 'z_is_aborted_timeout'? [-Wimplicit-function-declaration]
  139 |                 z_abort_timeout(&thread->base.timeout);
      |                 ^~~~~~~~~~~~~~~
      |                 z_is_aborted_timeout
.../zephyrproject/zephyr/kernel/events.c:139:46: error: 'struct _thread_base' has no member named 'timeout'
  139 |                 z_abort_timeout(&thread->base.timeout);
      |                                              ^
[124/131] Building C object zephyr/kernel/CMakeFiles/kernel.dir/dynamic_disabled.c.obj
ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

Linux
Zephyr SDK
main 67f246477f93ac8e7bd4b953a9f47d4f5f5b4ef5

### Additional Context

_No response_

--- a/kernel/events.c
+++ b/kernel/events.c
@@ -136,7 +136,9 @@ static int event_walk_op(struct k_thread *thread, void *data)
 		 * have been processed.
 		 */
 		thread->no_wake_on_timeout = true;
+#ifdef CONFIG_SYS_CLOCK_EXISTS
 		z_abort_timeout(&thread->base.timeout);
+#endif /* CONFIG_SYS_CLOCK_EXISTS */
 	}
 
 	return 0;

